mitmproxy/tools/console/common.py

```diff
@@ -67,6 +68,26 @@ def format_keyvals(
     if indent > 2:
         indent -= 2  # We use dividechars=2 below, which already adds two empty spaces
 
+    cols, _ = urwid.raw_display.Screen().get_cols_rows()
```

Comment:
I haven't tested this, but this looks like it will fall apart when using layouts that don't span the entire width. Can we pass in the available space instead?

Reply:
Hi @mhils ,
thanks for the review and the awesome work you people do here!

I'm not sure I'm following you: my (admittedly empirical) tests proved the fix to 
work on layouts smaller than full screen (see attached screenshot - header values are not wrapped even if the terminal size is half that of the screen, instead the header name is wrapped at 30 chars). Are there different scenarios you're thinking of?

Furthermore, I investigated a bit more and saw that mitmproxy already uses the same way to retrieve screen columns elsewhere, for instance in [mitmproxy/tools/console/statusbar.py](https://github.com/mitmproxy/mitmproxy/blob/4d00ec03b4fd0a187ed346d438395af3b5e3c3d5/mitmproxy/tools/console/statusbar.py#L81) (note that `self.master.ui` inherits from `urwid.raw_display.Screen` as it can be seen in [mitmproxy/tools/console/window.py](https://github.com/mitmproxy/mitmproxy/blob/4d00ec03b4fd0a187ed346d438395af3b5e3c3d5/mitmproxy/tools/console/window.py#L306).

Anyhow I am by no means an expert on `urwid`, so I'll be glad to refine the PR if you still think this approach is broken.

Screenshot:
![mitmproxy_pr](https://github.com/mitmproxy/mitmproxy/assets/115867985/6928ac63-4ba5-4983-a690-ed600735cc37)

Reply:
> Are there different scenarios you're thinking of?

\You can press `-` to toggle between views. If you enable the two column view, `urwid.raw_display.Screen().get_cols_rows()` will give you twice the space available:

![image](https://github.com/mitmproxy/mitmproxy/assets/1019198/a6f29d65-6754-42cc-a216-b17e30b098d0)

The statusbar "gets away" with this by always being full width.

After looking into this for a bit, the correct fix is unfortunately more complex. Urwid widgets famously have no widths, so we'd probably need to implement a custom widget here or something like that. :/ I haven't looked super close, but it looks a bit ugly.